client-spark/spark-3/src/main/scala/org/apache/spark/shuffle/celeborn/CelebornShuffleReader.scala

```diff
@@ -50,21 +49,7 @@ class CelebornShuffleReader[K, C](
 
   override def read(): Iterator[Product2[K, C]] = {
 
-    var serializerInstance = dep.serializer.newInstance()
-    if (conf.columnarShuffleEnabled) {
-      val schema = SparkUtils.getSchema(dep)
-      if (CelebornBatchBuilder.supportsColumnarType(
-          schema)) {
-        val dataSize = SparkUtils.getDataSize(
-          dep.serializer.asInstanceOf[UnsafeRowSerializer])
-        serializerInstance = new CelebornColumnarBatchSerializer(
-          schema,
-          conf.columnarShuffleBatchSize,
-          conf.columnarShuffleDictionaryEnabled,
-          conf.columnarShuffleOffHeapEnabled,
-          dataSize).newInstance()
-      }
-    }
+    val serializerInstance = newSerializerInstance(dep)
```

Comment:
Consider we merged #1838, we should rebase and tweak the logic below:

```scala
    val iterWithUpdatedRecordsRead =
      if (GlutenColumnarBatchSerdeHelper.isGlutenSerde(serializerInstance.getClass.getName)) {
        GlutenColumnarBatchSerdeHelper.withUpdatedRecordsRead(recordIter, metrics)
      } else {
        recordIter.map { record =>
          metrics.incRecordsRead(1)
          record
        }
      }
```

Reply:
Gluten Columnar Shuffle is different from Celeborn Columnar Shuffle.
It is clean and won't stop us to upgrade spark dependency version.
Let's leave it here for now.